src/main/java/com/google/devtools/build/lib/analysis/config/ToolchainTypeRequirement.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.analysis.config;
+
+import com.google.auto.value.AutoValue;
+import com.google.devtools.build.lib.cmdline.Label;
+
+/**
+ * Describes a requirement on a specific toolchain type.
+ */
+@AutoValue
+public abstract class ToolchainTypeRequirement {
+  /** Returns a builder for a new {@Link ToolchainTypeRequirement}. */
+  public static Builder builder() {
+    return new AutoValue_ToolchainTypeRequirement.Builder()
+        .mandatory(true);
+  }
+
+  /** The toolchain type this requirement is for. */
+  public abstract Label toolchainType();
+
+  /** Whether the toolchain type is mandatory or optional. */
```

Comment:
Can we contextualize this comment a bit more? Just another sentence suggesting the implication of these terms?

Reply:
Done.